extensions/amp-video/1.0/base-element.js

```diff
@@ -88,20 +88,19 @@ VideoBaseElement['props'] = {
   'artwork': {attr: 'artwork'},
   'attribution': {attr: 'attribution'},
   'autoplay': {attr: 'autoplay', type: 'boolean'},
-  'children': {
-    'sources': {
-      selector: 'source',
-      single: false,
-      clone: true,
-    },
-  },
+  'children': {},
```

Comment:
`'children: {}` is necessary here to tell BaseElement that this component uses the shadow. Alternatively, we can allow overriding `usesShadowDom` as a property of `BaseElement` and have the default value of that look for `['props']['children']`.

Reply:
I updated it to use `usesShadowDom`, but a large number of these components also have `shadowCss`, do you think it's possible to reuse this for a smart default value for `usesShadowDom`?

Reply:
I don't think we can. But certainly we can assert these values from each other. E.g. `devAssert(!shadowCss || usesShadowDom)` or such.